src/components/Navigation.js

```diff
@@ -22,6 +23,12 @@ export const Navigation = props => {
     setCurrent(pathNames[ props.location.pathname ]);
   }, [props.location.pathname]);
   
+  useEffect(() => {
```

Comment:
No need to call checkUserRegistered in this useEffect. It is always called on signedIn. So every time the user is signs in.